inc/http-client/class-cache-strategy.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace RemoteDataBlocks\HttpClient;
+
+use Kevinrob\GuzzleCache\Strategy\GreedyCacheStrategy;
+use Psr\Http\Message\RequestInterface;
+use RemoteDataBlocks\Logging\LoggerManager;
+
+class CacheStrategy extends GreedyCacheStrategy {
+	const HEADER_TTL          = 'X-RemoteDataBlocks-TTL';
+	const HEADER_BYPASS_CACHE = 'X-RemoteDataBlocks-Bypass-Cache';
+
+	private static function getRequestString( RequestInterface $request ) {
+		$uri        = $request->getUri();
+		$uri_string = $uri->getScheme() . '://' . $uri->getHost() . $uri->getPath();
+		$method     = $request->getMethod();
+		$body       = $request->getBody()->getContents();
+		return $method . ' ' . $uri_string . ' ' . $body;
+	}
+
+	protected function should_bypass_cache( RequestInterface $request ) {
+		$bypass = $request->getHeader( self::HEADER_BYPASS_CACHE );
```

Comment:
We can inspect the request here to e.g. put tighter scrutiny on `POST` requests.

Reply:
e.g. We discussed inspecting the GraphQL operation for mutations and setting those to not cache.

Additionally, we discussed using a method on e.g. the `QueryContext` that we leverage to bypass the cache.